### PR TITLE
Feature/performance

### PR DESCRIPTION
**Pull request by klzns**
#### What is the purpose of this pull request?

Improve performance

#### What problem is this solving?

- Keep only entry-points in root of `react/`
- Remove dependency
- Use function components instead of classes

#### How should this be manually tested?

https://breno--storecomponents.myvtexdev.com/?disableUserLand

#### Screenshots or example usage

Before |
---|
image |

After |
---|
image | 
 
#### Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ShelfItem.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import normalizeProduct from '../utils/normalizeProduct'
+
+/**
+ * ShelfItem Component. Normalizes the item received in the props
+ * to adapt to the extension point prop.
+ */
+const ShelfItem = ({ item, position, summary }) => {
+  const { push } = usePixel()
+  const newSummary = useMemo(
+    () => assocPath(['name', 'tag'], 'h2', summary),
```
Comment (lbebber) [inline react/components/ShelfItem.js]: ☹️

File: react/components/ShelfItem.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      event: 'productClick',
+      product: product,
+    })
+  }, [product, push])
+
+  return (
+    <ExtensionPoint
+      id="product-summary"
+      product={product}
+      actionOnClick={pushPixelProductClick}
```
Comment (lbebber) [inline react/components/ShelfItem.js]: quem tem prop `actionOnClick`?

File: react/package.json [hunk 3]
```diff
@@ -15,8 +15,7 @@
     "classnames": "^2.2.6",
     "ramda": "^0.25.0",
     "react-intl": "^2.7.2",
-    "react-resize-detector": "^3.4.0",
-    "recompose": "^0.30.0"
```
Comment (lbebber) [inline react/package.json]: 🙏

File: react/components/ShelfItem.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import { path, assocPath } from 'ramda'
+
+import { ExtensionPoint } from 'vtex.render-runtime'
+import { usePixel } from 'vtex.pixel-manager/PixelContext'
+
+import { shelfItemPropTypes } from '../utils/propTypes'
+import normalizeProduct from '../utils/normalizeProduct'
+
+/**
+ * ShelfItem Component. Normalizes the item received in the props
```
Comment (lbebber) [inline react/components/ShelfItem.js]:
```suggestion
 * Normalizes the item received in the props
```

Comment (lbebber) [reply react/components/ShelfItem.js]: (sugestao)

File: react/components/ShelfItem.js (hunk 2 above)
Comment (klzns) [reply react/components/ShelfItem.js]: Product Summary

**Pull request merged**